controller/controller.go

```diff
@@ -206,6 +253,11 @@ func (c controller) GetStatus(req Request) ([]fleet.UnitStatus, error) {
 	return status, maskAny(err)
 }
 
+func (c controller) WaitForTask(taskObject *task.TaskObject, closer <-chan struct{}) (*task.TaskObject, error) {
```

Comment:
> and the task service

Why? The taskservice returns a struct which also fulfills the interface the controller defines and returns.